### ISSUE TITLE
Deep nesting of string interpolation slows parser

### ISSUE THREAD

**Issue by itchyny** [MiSawa/xq#37]
Nesting string interpolation extremely slows the parser (not execution because adding invalid token to the query does not exit quickly).
```sh
❯ time xq -n '"\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\(1+2)")")")")")")")")")")")")")")")")")")")")")"'
"3"
xq -n   16.01s user 0.02s system 99% cpu 16.034 total

❯ time jq -n '"\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\("\(1+2)")")")")")")")")")")")")")")")")")")")")")"'
"3"
jq -n   0.00s user 0.00s system 65% cpu 0.010 total
```

**Comment by MiSawa** [comment]
Thank you for reporting!
Seems to be explosion of `alt`, though not sure which component of parser is causing it.

Perhaps it could be resolved by adding `cut`s like [this example](https://github.com/Geal/nom/blob/e99f9e0e7e8ba5670ecc4c0dc89b1d90ff7ba865/examples/json.rs), though wasn't able to locate appropriate parts to add to yet.
Or maybe #2 is easier.

**Comment by MiSawa** [comment]
Actually it probably isn't as easy as adding cuts. Need to figure out what branch is causing the issue.

**Comment by MiSawa** [comment]
Fixed in #52
